infrastructure/lib/atat-iam-stack.ts

```diff
@@ -0,0 +1,184 @@
+  private readonly outputs: cdk.CfnOutput[] = [];
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+    this.templateOptions.description = "Creates the IAM permissions for ATAT developers, testers, and auditors.";
+
+    // job-function/ReadOnlyAccess is actually a somewhat-dangerous IAM policy that
+    // allows users to view sensitive data (such as Lambda environment variables or
+    // some Secrets Manager Secrets). ViewOnlyAccess is more restrictive.
+    // All roles will have ViewOnly access.
+    const awsManagedViewOnlyPolicy = iam.ManagedPolicy.fromAwsManagedPolicyName("job-function/ViewOnlyAccess");
+    const awsManagedLogsReadPolicy = iam.ManagedPolicy.fromAwsManagedPolicyName("CloudWatchLogsReadOnlyAccess");
+    const awsManagedAuditorPolicy = iam.ManagedPolicy.fromAwsManagedPolicyName("SecurityAudit");
+    const awsManagedAdminAccessPolicy = iam.ManagedPolicy.fromAwsManagedPolicyName("AdministratorAccess");
+    const awsManagedCfnFullAccessPolicy = iam.ManagedPolicy.fromAwsManagedPolicyName("AWSCloudFormationFullAccess");
+
+    // In the future, we may want to consider moving to a mechanism where the various
+    // resources, when created, grant access to these administrative roles rather than
+    // having us globally grant access at this level. For example, at this point, it
+    // is appropriate for our users to have access to read all tables directly but
+    // will it always? Might we eventually store sensitive data there?
+    // Eventually, many of these actions may require access to a KMS Key.
+    const generalReadAccess = new iam.ManagedPolicy(this, "GeneralReadAccess", {
+      statements: [
+        new iam.PolicyStatement({
+          sid: "DynamoDBItemAccess",
+          effect: iam.Effect.ALLOW,
+          actions: ["dynamodb:*GetItem", "dynamodb:PartiQLSelect", "dynamodb:Scan", "dynamodb:Query"],
+          resources: [`arn:${cdk.Aws.PARTITION}:dynamodb:*:${cdk.Aws.ACCOUNT_ID}:table/*`],
+        }),
+        new iam.PolicyStatement({
+          sid: "APIGatewayRestApiReadAccess",
+          effect: iam.Effect.ALLOW,
+          actions: ["apigateway:GET"],
+          resources: [`arn:${cdk.Aws.PARTITION}:apigateway:*::/restapis*`],
+        }),
+      ],
+    });
+
+    const baseDenies = new iam.ManagedPolicy(this, "AtatUserDenyPolicy", {
+      statements: [
+        new iam.PolicyStatement({
+          sid: "DenyAllOrganizations",
+          effect: iam.Effect.DENY,
+          actions: ["organizations:*"],
+          resources: ["*"],
+        }),
+      ],
+    });
+
+    const developerCdkStagingAccess = new iam.ManagedPolicy(this, "DeveloperCdkStagingAccess", {
+      statements: [
+        new iam.PolicyStatement({
+          sid: "AllowModifyingCdkToolBuckets",
+          effect: iam.Effect.ALLOW,
+          actions: ["s3:*"],
+          resources: [
+            `arn:${cdk.Aws.PARTITION}:s3:::cdk-hnb659fds-assets-${cdk.Aws.ACCOUNT_ID}-*`,
```

Comment:
Intentionally hard-coded ID here? Could you add a comment describing what it is?

Reply:
yeah so that's a great point. from: https://docs.aws.amazon.com/cdk/latest/guide/bootstrapping.html

> `--qualifier` a string that is added to the names of all resources in the bootstrap stack. A qualifier lets you avoid name clashes when you provision two bootstrap stacks in the same environment. The default is `hnb659fds` (this value has no significance). Changing the qualifier will require changes to your AWS CDK app (see Stack synthesizers).

There might actually be a better way to get this value. I'll look into it.

Reply:
Maybe including a comment with that URL is good enough.

Reply:
Going to just change this to `cdk-*-assets` just in case we use other qualifiers in the future. And I actually will review Access Analyzer to see if this line of the policy ever gets used once it's in place. We may really only need the staging bucket.